website/pages/docs/agent/template/index.mdx

```diff
@@ -124,3 +124,31 @@ template {
 If you only want to use the Vault agent to render one or more templates and do
 not need to sink the acquired credentials, you can omit the `sink` stanza from
 the `auto_auth` stanza in the agent configuration.
+
+## Renewals and Updating Secrets
+
+The Vault Agent templating automatically renews and fetches secrets.  The behavior
+of how Agent does this depends on the type of secret.  The following is a high level
+overview of different behaviors.
+
+### Renewable Secrets
+
+If a secret is renewable, Vault Agent will renew and fetch the secret when 85% of the
+secrets time-to-live (TTL) is reached. Renewable secrets include (but not limited
+to) tokens, dynamic secrets such as [database credentials](/docs/secrets/databases) and [KV Version 1](/docs/secrets/kv/kv-v1).
+
+### Non-Renewable Secrets
+
+If a secret isn't renewable, Vault Agent will fetch the secret every 5 minutes. This is not
+configurable.  Non-renewable secrets include (but not limited to) [KV Version 2](/docs/secrets/kv/kv-v2).
+
+### Static Roles
+
+If a secret has a `rotation_period`, such as a [database static role](/docs/secrets/databases#static-roles),
+Vault Agent template will fetch the new secret as it changes in Vault.  It does
+this by inspecting the secret's time-to-live (TTL).
+
+### Certificates
```

Comment:
Do we want to call out if that if `generate_lease` is used certs will fall under the "renewable" category (well it's not renewable, but it does have a leaseID...)

Reply:
Good point, fixed!